File: leetcode/easy/1800-maximum-ascending-subarray-sum/solution.php
<?php

class Solution
{
    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function maxAscendingSum($nums)
    {
        $maxSum = $currentSum = $nums[0];
        for ($i = 1; $i < count($nums); $i++) {
            if ($nums[$i] > $nums[$i - 1]) {
                $currentSum += $nums[$i];
            } else {
                $maxSum = max($maxSum, $currentSum);
                $currentSum = $nums[$i];
            }
        }
        return max($maxSum, $currentSum);
    }
}

$solution = new Solution();
$nums = [10, 20, 30, 5, 10, 50];
echo $solution->maxAscendingSum($nums); // Output: 65

File: leetcode/easy/1800-maximum-ascending-subarray-sum/solution.rs
use std::cmp::max;

impl Solution {
    pub fn max_ascending_sum(nums: Vec<i32>) -> i32 {
        let mut max_sum = nums[0];
        let mut current_sum = nums[0];
        for i in 1..nums.len() {
            if nums[i] > nums[i - 1] {
                current_sum += nums[i];
            } else {
                max_sum = max(max_sum, current_sum);
                current_sum = nums[i];
            }
        }
        max(max_sum, current_sum)
    }
}
